### ISSUE TITLE
teetty passthrough SIGWINCH support

### ISSUE THREAD

**Issue by mullikine** [mitsuhiko/teetty#8]
It would be really awesome if teetty would passthrough the SIGWINCH signal to the inferior pty. i.e. capture the SIGWINCH signal sent to teetty and send it to the inferior process of the teetty pty. Or perhaps this is the job of another utility to proxy signals.

Even tcl/expect does not support this, to my knowledge.

http://www.rkoucha.fr/tech_corner/sigwinch.html

Example of resizing inside and out of teetty:

https://asciinema.org/a/548055

**Comment by mitsuhiko** [comment]
It is actually trying to do that but for some reason failing. This is a bug I haven't been able to fix yet.

https://github.com/mitsuhiko/teetty/blob/99440fc98aa934c2b52d2a9f07294b3a1f08278d/src/spawn.rs#L162-L179

**Comment by mitsuhiko** [comment]
If someone wants to debug this. Here is what I tried to do so far:

This Python script prints out on window size changes the current terminal size:

```python
import time
import struct, fcntl, termios, signal, sys

def sigwinch_passthrough (sig, data):
    s = struct.pack("HHHH", 0, 0, 0, 0)
    a = struct.unpack('hhhh', fcntl.ioctl(sys.stdout.fileno(),
        termios.TIOCGWINSZ , s))
    print(a[0], a[1])

signal.signal(signal.SIGWINCH, sigwinch_passthrough)

while True:
    time.sleep(1)
```

When run in teetty it always prints out the initial terminal size. The `get_winsize` call however in the `sigwinch_passthrough` prints out the right information. Likewise the `wet_winsize` call to the pty also does not create an error and forwards the right window size. Likewise I already verified that `killpg` is going to the right process. Without it Python does not see the signal arrive.

So somehow the `TIOCGWINSZ` `ioctl` write to the terminal has no effect.
